magicmaze.js

```diff
@@ -657,21 +657,55 @@ function drawMagePreviews (obj) {
   }
 }
 
-function placeCharacter (obj, info) {
+
+
+function placeCharacter (obj, info, warp = false) {
   const x = parseInt(info.position_x)
   const y = parseInt(info.position_y)
   const tokenId = parseInt(info.token_id)
+  const [oldx, oldy] = obj.characterLocs.get(tokenId) ?? [x, y]
   obj.characterLocs.set(tokenId, [x, y])
   const key = getKey(x, y)
   const top = obj.tops.get(key)
   const left = obj.lefts.get(key)
   const adjust = (CELL_SIZE - MEEPLE_SIZE) / 2
+  var halftime = 100
+  if (warp) {
+    halftime = 300
+  }
   obj.rescale(1.0)
-  obj.slideToObjectPos(`mm_token${info.token_id}`,
+  const slide = obj.slideToObjectPos(
+    `mm_token${info.token_id}`,
     'mm_area_scrollable_oversurface',
     left + adjust,
     top + adjust,
-    /* duration */ 200).play()
+    2*halftime)
+  slide.play()
```

Comment:
oh I think we should also probably start this _after_ we hook up the onEnd hook, but I forgot to comment about this because I'm still confused about why "onEnd" is at a very weird time